articles/introduction-to-avl-trees/index.md

```diff
@@ -0,0 +1,578 @@
+        return (newNode(val));
+
+    // If the value to be inserted is lesser than the current node, traverse towards the left
+    if (val < node->value)
+       {
+            node->leftChild = insert_AVL(node->leftChild, val);
+            // Balance the nodes after insertion
+            node=balance(node);d
+       }
+    
+    // If the value to be inserted is greater than the current node, traverse towards the right
+    if (val > node->value)
+        {
+            node->rightChild = insert_AVL(node->rightChild, val);
+            // Balance the nodes after insertion
+            node= balance(node);
+        }
+    else
+        return node;
+}
+
+void inOrder(Node *root)  
+{  
+    if(root != NULL)  
+    {  
+        inOrder(root->leftChild); 
+        cout << root->value << " ";  
+        inOrder(root->rightChild);  
+    }  
+} 
+
+// Driver Code
+int main(void)
+{
+    Node *root = NULL;  
+    root = insert_AVL(root, 1);  
+    cout << "The inorder traversal of the AVL tree is: "<<endl;  
+    inOrder(root);
+    // Output - The inorder traversal of the AVL tree is: 1
+    cout << endl;
+    root = insert_AVL(root, 3);
+    cout << "The inorder traversal of the AVL tree is: "<<endl;  
+    inOrder(root);
+    cout << endl;
+    // Output - The inorder traversal of the AVL tree is: 1 3
+    root = insert_AVL(root, 4); 
+    cout << "The inorder traversal of the AVL tree is: "<<endl;  
+    inOrder(root);
+    cout << endl;
+    // Output - The inorder traversal of the AVL tree is: 1 3 4
+    root = insert_AVL(root, 5); 
+    cout << "The inorder traversal of the AVL tree is: "<<endl;  
+    inOrder(root);
+    cout << endl;
+    // Output - The inorder traversal of the AVL tree is: 1 3 4 5
+    root = insert_AVL(root, 6);
+    cout << "The inorder traversal of the AVL tree is: "<<endl;  
+    inOrder(root);
+    cout << endl;
+    // Output - The inorder traversal of the AVL tree is: 1 3 4 5 6
+    root = insert_AVL(root, 2);  
+    cout << "The inorder traversal of the AVL tree is: "<<endl;  
+    inOrder(root);  
+    cout<<endl;
+    // Output - The inorder traversal of the AVL tree is: 1 2 3 4 5 6       
+}
+```
+
+Since the tree is 'automatically' balanced after every insertion, the complexity of insertion remains at O(logn), unlike a regular BST.
+
+### Deletion from AVL Trees in C++
+
+Deletion follows a very similar algorithm compared to insertion. However, one must keep in mind that different nodes can be deleted i.e. nodes with one child, nodes with two children and nodes with no children. These cases have to be taken into account while balancing the tree.
+
+    1. Perform regular BST deletion.
+    2. Get the balance factor of the current node (height of left subtree - height of right subtree).
+    3. If the balance factor is equal to 2, then the case could be either right-right or right-left rotation. If the balance of the left tree is equal to 1, then right-right rotation, else right-left rotation.
```

Comment:
There are quite a few errors in the algorithm you've suggested for AVL tree deletion.

Please refer to https://www.geeksforgeeks.org/avl-tree-set-2-deletion/ and check the algorithm once.